File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/ApiResponseModels/ResponseMessageModel.cs
using KnowledgeSharingApi.Domains.Models.Entities;
using KnowledgeSharingApi.Domains.Models.Entities.Tables;
using KnowledgeSharingApi.Domains.Models.Entities.Views;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.ApiResponseModels
{
    public class ResponseMessageModel : ViewMessage
    {
        public ResponseMessageModel? ReplyMessage { get; set; } = null;


        protected override ResponseMessageModel Init()
        {
            return new ResponseMessageModel();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/api-response-models/response-message-model.js

import ViewMessage from "../views/view-message";

class ResponseMessageModel extends ViewMessage {
    constructor() {
        super();
        this.ReplyMessage = null;
    }

    init() {
        return new ResponseMessageModel();
    }

    copy(entity){
        super.copy(entity);
        if (this.ReplyMessage != null) {
            this.ReplyMessage = new ResponseMessageModel().copy(this.ReplyMessage);
        }
        return this;
    }
}

export default ResponseMessageModel;
